### PR TITLE
Lucky and Unlucky traits (and quirks); affects chance of being selected for events

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Added the Lucky and Unlucky quirks. Having these will either halve or double your chances of being selected
for a negative or positive event that affects a single person, such as Spontaneous Appendicitis, Disease
Outbreak or a random Heart Attack. It does not affect any other form of RNG.
/:cl:

I thought this would be a neat thing to code. It'd be interesting how much people value being more likely
to be targeted by negative events. We don't actually have any positive RNG events that I'm aware of (please tell
me if I missed anything).

If and when Spontaneous Brain Trauma is added, this'd be covered by the luck traits.

### CODE REVIEW

**Comment by BeeSting12** [comment]
Can this transfer to when youre a ghost

**Comment by XDTM** [comment]
Random disease event

**Comment by XDTM** [comment]
Some positive events would be neat, but events are meant to disrupt the round, so i wonder what kind of positive event could throw a wrench in the station's works

**Comment by tralezab** [comment]
I'll pr some positive events, I have a couple ideas.

**Comment by Ratvar** [comment]
...some negative events will affect multiple people, example - random virus infection. Cue the assistants spreading killcough thrice as much, griefing by passive that gives them more graytide points?

**Comment by XDTM** [comment]
Yeah but it's not like you get to choose which bad event you want. Most are just bad for the victim.

File: code/__HELPERS/mobs.dm [hunk 1]
```diff
@@ -458,3 +458,31 @@ GLOBAL_LIST_EMPTY(species_list)
...
+
+			if(type == NEGATIVE_EVENT)
+				if(prob(50))
+					applied_luck += M
+
+		else if(M.has_trait(TRAIT_UNLUCKY))
+			if(type == POSITIVE_EVENT)
+				if(prob(50))
+					applied_luck += M
+			if(type == NEGATIVE_EVENT)
```
Comment (Cyberboss) [inline code/__HELPERS/mobs.dm]:
```suggestion
			else if(type == NEGATIVE_EVENT)
```

File: code/__HELPERS/mobs.dm [hunk 2]
```diff
@@ -458,3 +458,31 @@ GLOBAL_LIST_EMPTY(species_list)
...
+			applied_luck += M
+		// Good and bad luck cancel each other out.
+		else if(M.has_trait(TRAIT_LUCKY) && M.has_trait(TRAIT_UNLUCKY))
+			applied_luck += M
+
+		else if(M.has_trait(TRAIT_LUCKY))
+			if(type == POSITIVE_EVENT)
+				applied_luck += list(M, M)
+
+			if(type == NEGATIVE_EVENT)
```
Comment (Cyberboss) [inline code/__HELPERS/mobs.dm]:
```suggestion
			else if(type == NEGATIVE_EVENT)
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/datums/traits/good.dm [hunk 3]
```diff
@@ -148,3 +148,9 @@
 	mob_trait = TRAIT_VORACIOUS
 	gain_text = "<span class='notice'>You feel HONGRY.</span>"
 	lose_text = "<span class='danger'>You no longer feel HONGRY.</span>"
+
+/datum/quirk/lucky
```
Comment (Mickyan) [inline code/datums/traits/good.dm]: put this in alphabetic order

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]

**Comment by optimumtact** [comment]
I'm very strongly against affecting roll chances for all jobs, antag statues and events, so as such, I'm denying this, in the interests of keeping the game fair for people without understanding of all game mechanics.

**Pull request closed without merging**